SCRUM-16:Get_Email_Important

ErickaIsa opened the pull request:
Se añadio test cases a la carpeta de test y los aserts con un bug encontrado al comparar el schema al compar con el Json
![image_2024-06-29_221341161](https://github.com/Jasonn5/Qates/assets/55772207/e49ba3b4-45e0-4185-8113-63af082bc3b5)
![image](https://github.com/Jasonn5/Qates/assets/55772207/ba7905a5-f264-432f-b5fd-6e5c9440dd1b)
![image](https://github.com/Jasonn5/Qates/assets/55772207/12f60f8e-d0f8-4049-b387-5930b6cfff09)
![image](https://github.com/Jasonn5/Qates/assets/55772207/ba42ed94-4dfa-47f0-ac64-3dc63dbdbd86)
![image](https://github.com/Jasonn5/Qates/assets/55772207/66e561f3-e745-459b-8c25-bf2e77baf735)

Jasonn5 committed:
Initial commit

Jasonn5 committed:
login-Jeyson

Jasonn5 committed:
delete error

Jasonn5 committed:
login-Jeyson

Jasonn5 committed:
refactor

Jasonn5 committed:
Refactor Login

AlisonGuzman2006 committed:
Merge pull request #2 from Jasonn5/Login-Jeyson

Login-Jeyson

Jasonn5 committed:
Get_Preentations

Add method to automations to test next test cases:
-Verify that the endpoint returns a successful response (status code 200)
-Successfully get meetings and validate the JSON schema
-Verify that the response is in JSON format
-Verify that meetings cannot be obtained without authentication

Jasonn5 committed:
Add postman response

Jasonn5 committed:
Restructure Architecture

AlisonGuzman2006 committed:
Merge pull request #8 from Jasonn5/Resturucture_Archirecture

Resturucture Archirecture

Jasonn5 committed:
Get_Presentations

Added automations to next test cases:
-Verify that the endpoint returns a successful response (status code 200)
-Successfully get meetings and validate the JSON schema
-Verify that the response is in JSON format
-Verify that meetings cannot be obtained without authentication

AlisonGuzman2006 committed:
Merge pull request #9 from Jasonn5/Get_Presentacions

We confirm the merge request of the branch Get_Presentations to the branch development

ErickaIsa committed:
SCRUM-16:Post_Email_Send email

se añadio test cases a la carpeta de test y los aserts

ErickaIsa committed:
SCRUM-16:Post_Email_Send email

se añadio test cases a la carpeta de test y los aserts

AlyGASS committed:
SCRUM-5/S RAgregar ls métodos de assertion en su apartado correspondiente de la estructura del proyecto para el endpoint getCalls

FreddyFZ17 commented on tests/correoImportant/test_GetCorreoImportant.py:
With respect to the following methods "test_get_email_succed and test_get_email_response_format" I have no observations, since they were executed without problem and in my opinion they are well integrated, but with respect to test_get_email_schema_validation I had a problem when executing said testcase, this being that there is a validation problem of the JSON schema when checking the response received from the API.

AlisonGuzman2006 replied:
Please review you test, and make sure that they are using you email param properly, as well as you enum.

AlisonGuzman2006 replied:
I approve the corrections made

FreddyFZ17 commented on src/resources/schemas/correoImportant.json:
Considering that the root of the problem arises from the schema, I think that actions should be taken on said json schema.

AlisonGuzman2006 replied:
This has to be corrected, because the enum class created for this endpoint is not being used properly

AlisonGuzman2006 replied:
I aprove the corrections made

AlyGASS committed:
S
SCRUM-5: Get_All_Calls
Implemented Test Cases to verify the following scenarios:

- test_get_allCalls_withNoParams_success
- test_get_calls_schema_validation
- test_get_allCalls_withNoParams_format
- test_get_calls_invalid_cookie_in_headers
- test_get_calls_invalid_authorization_in_headers
- test_get_calls_with_param_select_maxsize_offset_orderby_otest_get_calls_with_param_select_otherValidData
- test_get_calls_with_param_select_invalidData
- test_get_calls_with_param_select_empty
- test_get_calls_with_param_maxsize_otherValidData
- test_get_calls_with_param_maxsize_invalidData
- test_get_calls_with_param_offset_otherValidData
- test_get_calls_with_param_offset_invalidData
- test_get_calls_with_param_offset_empty
- test_get_calls_with_param_orderBy_otherValidData
- test_get_calls_with_param_orderBY_invalidData
- test_get_calls_with_param_orderBy_empty

Added detailed print statements for debugging and validation:
  - Print request headers, URL, response status code, and response text for selected test cases.

Note: While most tests passed, a few tests returned unexpectedly.

AlyGASS committed:
SCRUM-15: Get_All_Calls
Re-structured the proyect by adding the api_request and calls_endpoint classes

- api_request.py: Normal class, that has different Get methods based on the different param set to the get like: url, header and/or parameters used in the get:
	- get_with_url(self, url)
	- get_with_url_headers(self, url, headers)
	- get_with_url_headers_params(self, url, headers, params)

- calls_endpoints.py: Enum class specifically used for the getCalls endpoint, this class has the differents endpoints used by the get. This can be used as an example for the other team members.
	- GET_CALLS_WITHPUT_PARAMS
	- GET_CALLS_WITH_PARAMS

Note: This new structure must be applied to all other endpoint that are being tested.

AlyGASS committed:
Refractored global variable calls_param to CALL_PARAM and the file assert_headers to assertion_headers

ErickaIsa committed:
Add API test cases for invalid order parameter, valid maxSize, missing auth token, valid offset, and invalid basic authorization

- Added test for invalid order parameter resulting in 400 status code
- Added test for valid maxSize parameter ensuring successful response
- Added test for missing auth token resulting in 401 unauthorized status
- Added test for valid offset parameter ensuring successful response
- Added test for invalid basic authorization resulting in 401 unauthorized status

ErickaIsa committed:
Add API test cases for invalid order parameter, valid maxSize, missing auth token, valid offset, and invalid basic authorization

- Added test for invalid order parameter resulting in 400 status code
- Added test for valid maxSize parameter ensuring successful response
- Added test for missing auth token resulting in 401 unauthorized status
- Added test for valid offset parameter ensuring successful response
- Added test for invalid basic authorization resulting in 401 unauthorized status

ErickaIsa committed:
Add API test cases for invalid order parameter, valid maxSize, missing auth token, valid offset, and invalid basic authorization

- Added test for invalid order parameter resulting in 400 status code
- Added test for valid maxSize parameter ensuring successful response
- Added test for missing auth token resulting in 401 unauthorized status
- Added test for valid offset parameter ensuring successful response
- Added test for invalid basic authorization resulting in 401 unauthorized status2

ErickaIsa committed:
Add API test cases for invalid order parameter, valid maxSize, missing auth token, valid offset, and invalid basic authorization

- Added test for invalid order parameter resulting in 400 status code
- Added test for valid maxSize parameter ensuring successful response
- Added test for missing auth token resulting in 401 unauthorized status
- Added test for valid offset parameter ensuring successful response
- Added test for invalid basic authorization resulting in 401 unauthorized status2

Jasonn5 committed:
Merge pull request #12 from Jasonn5/SCRUM-5_Get_All_Calls

SCRUM-5/S RAgregar ls métodos de assertion en su apartado correspondi…

ErickaIsa committed:
Merge branch 'development' of https://github.com/Jasonn5/Qates into development

ErickaIsa committed:
edit

ErickaIsa committed:
resolviendo conflictos

ErickaIsa committed:
Add tests for 'Important Emails' API endpoint

- Implemented tests for the 'Important Emails' API endpoint to ensure proper functionality.
- Added validation for the response schema to verify the structure of the JSON response.
- Included tests to check the response format (application/json).
- Tested invalid cookie and authorization scenarios to verify they return the correct unauthorized status.
- Added tests for various query parameters to ensure the API handles them correctly.
- Ensured proper handling of missing authentication tokens and invalid authorization credentials.
- Improved overall test coverage and robustness for the 'Important Emails' endpoint.

ErickaIsa committed:
Merge branch 'development' of https://github.com/Jasonn5/Qates into development

ErickaIsa committed:
resolviendo conflictos 2

ErickaIsa committed:
Añadi marks a los tests en getCorreo.py

ErickaIsa committed:
comit x

ErickaIsa committed:
Merge branch 'development' of https://github.com/Jasonn5/Qates into development

ErickaIsa committed:
resolucion de conflictos

AlisonGuzman2006 commented on config.py:
The global variable for the Email, is accepted.

AlisonGuzman2006 commented on src/assertions/assertion_schemas.py:
This assertion method is also aproved, it has clean code

AlisonGuzman2006 commented on :
This has beed accepted.

ErickaIsa committed:
ediccion de codigo para usar la carpeta  enum

ErickaIsa committed:
cambios en el emal_Impotant_param

AlisonGuzman2006 reviewed (approved):
The changes and corrections have been accepted
